src/main/scala/com/fulcrumgenomics/umi/CodecConsensusCaller.scala

```diff
@@ -0,0 +1,196 @@
+
+/**
+  * TODO: write docs
+  *
+  * @param readNamePrefix the prefix to apply to all consensus read names
+  * @param readGroupId    the read group ID to apply to all created consensus reads
+  * @param minInputBaseQuality the minimum input base quality score to use a raw read's base
+  * @param errorRatePreUmi the estimated rate of errors in the DNA prior to attaching UMIs
+  * @param errorRatePostUmi the estimated rate of errors in the DNA post attaching UMIs
+  * @param minReadsPerStrand
+  * @param maxReads
+  * @param minDuplexLength
+  */
+class CodecConsensusCaller(readNamePrefix: String,
+                           readGroupId: String = "A",
+                           minInputBaseQuality: PhredScore = DuplexConsensusCaller.MinInputBaseQuality,
+                           errorRatePreUmi: PhredScore = DuplexConsensusCaller.ErrorRatePreUmi,
+                           errorRatePostUmi: PhredScore = DuplexConsensusCaller.ErrorRatePostUmi,
+                           val minReadsPerStrand: Int = 1,
+                           maxReadsPerStrand: Int = VanillaUmiConsensusCallerOptions.DefaultMaxReads,
+                           minDuplexLength: Int = 1
+                          ) extends DuplexConsensusCaller(
+  readNamePrefix      = readNamePrefix,
+  readGroupId         = readGroupId,
+  minInputBaseQuality = minInputBaseQuality,
+  qualityTrim         = false,
+  errorRatePreUmi     = errorRatePreUmi,
+  errorRatePostUmi    = errorRatePostUmi,
+  minReads            = Seq(1, 1, 1),
+  maxReadsPerStrand   = maxReadsPerStrand
+) {
+  require(this.minReadsPerStrand >= 1, "minReadsPerStrand must be at least 1.")
+
+  private val FilterDuplexTooShort = s"Top/Bottom Strand Overlap < ${minDuplexLength} bases."
+  private val FilterIndelError = "Indel error between strands in duplex overlap."
+  private val clipper = new SamRecordClipper(ClippingMode.Hard, autoClipAttributes = false)
+
+  /** Returns the MI tag as is because in CODEC there is no /A or /B unlike classic duplex-seq */
+  override protected[umi] def sourceMoleculeId(rec: SamRecord): String = rec[String]("MI")
+
+  /**
+    * Takes in all the reads for a source molecule and, if possible, generates one or more
+    * output consensus reads as SAM records.
+    *
+    * @param recs the full set of source SamRecords for a source molecule
+    * @return a seq of consensus SAM records, may be empty
+    */
+  override protected def consensusSamRecordsFromSamRecords(recs: Seq[SamRecord]): Seq[SamRecord] = {
+    val (pairs, frags) = recs.partition(_.paired)
+    rejectRecords(frags, FilterFragments)
+
+    if (pairs.isEmpty) {
+      Nil
+    }
+    else {
+      // Extract just the primary alignments and then clip where they extend past the ends of their mates
+      // TODO: Remove the isFrPair here and handle chimeric reads or reads with one unmapped etc.
+      val primaries = pairs.filterNot(r => r.secondary || r.supplementary).filter(_.isFrPair)
```

Comment:
question: Most aligners prefer to to have short inserts between R1 and R2, so this may not be an issue, but do we want to mention in the tool docs that we expect R1 and R2 to be almost fully overlapping, so alignment settings should be changed (e.g. insert size penalty) to really prefer overlapping alignments?

Reply:
I think this only happens in repetitive regions right?  I don't think I have enough experience with CODEC yet to be offering advice on alignment ... and I think I'd prefer to just not analyze reads in regions where the alignments are not of high confidence (after consensus calling).